### ISSUE TITLE
kha.Image.at returns incorrect color on iOS target

### ISSUE THREAD

**Issue by Joncom** [Kode/Kha#1000]
When reading from a `0x016F52` colored image using `kha.Image.at`, I expect the result to be `0xFF016F52`. However, I'm getting `0xFF526F01` instead. It would appear the 1st and 3rd least significant bytes (the red and blue bytes) are getting swapped for some reason.

**Comment by Joncom** [comment]
I wonder if this is related (quote from https://bjango.com/articles/pngcompression/):

> In an effort to dramatically increase drawing performance of iOS apps, Xcode re-compresses PNG files as it builds. It premultiplies the alpha channel and byte swaps the red, green and blue channels to be sequenced blue, green and red.

**Comment by RobDangerous** [comment]
No, it isn't, khamake prevents that png madness which Xcode tries to do. https://github.com/Kode/Kore/blob/master/Sources/Kore/Graphics1/Image.cpp#L325 just needs some more logic for different formats.

**Comment by Joncom** [comment]
![screen shot 2019-02-27 at 5 10 46 pm](https://user-images.githubusercontent.com/1667179/53483010-1953f380-3ab3-11e9-810e-b42c550d5585.png)

The format appears to be `RGBA32` and seems to be pulling the correct bytes already (just in the wrong order). The byte order appears to be ABGR instead of RGBA.

@RobDangerous, what extra logic are you suggesting? Could it be as simple as reversing the byte order before returning?

I wonder if this issue affects all Kore targets the same way?

**Comment by Joncom** [comment]
Just tested Android-native and it seems to be affected as well. Going to update the title of this issue accordingly.

**Comment by Joncom** [comment]
The following patch fixed the issue for me on both iOS and android-native:

```diff
diff --git a/Kha/Backends/Kore/kha/Image.hx b/Kha/Backends/Kore/kha/Image.hx
index b011dd1a..d67a3c98 100644
--- a/Kha/Backends/Kore/kha/Image.hx
+++ b/Kha/Backends/Kore/kha/Image.hx
@@ -292,7 +292,21 @@ class Image implements Canvas implements Resource {
                return true;
        }

-       @:functionCode('return texture->at(x, y);')
+       @:functionCode('
+               int color = texture->at(x, y);
+
+               unsigned char *r = (unsigned char *)&color;
+               // unsigned char *g = (unsigned char *)&color + 1;
+               unsigned char *b = (unsigned char *)&color + 2;
+               // unsigned char *a = (unsigned char *)&color + 3;
+
+               // Not sure why, but red and blue bytes are swapped, so swap them back.
+               unsigned char temp = *r;
+               *r = *b;
+               *b = temp;
+
+               return color;
+       ')
        private function atInternal(x: Int, y: Int): Int {
                return 0;
        }
```

**Comment by Joncom** [comment]
@RobDangerous, would it be OK to send this up as a pull-request? I'm still not really sure about this issue as a whole. I have no idea why the red and blue bytes are swapped...
